pyls/language_server.py

```diff
@@ -75,22 +77,28 @@ class LanguageServer(MethodJSONRPCServer):
     """
 
     process_id = None
-    root_path = None
+    root_uri = None
     init_opts = None
 
     def capabilities(self):
         return {}
 
-    def initialize(self, root_path, init_opts, process_id):
+    def initialize(self, root_uri, init_opts, process_id):
         pass
 
     def m_initialize(self, **kwargs):
         log.debug("Language server intialized with %s", kwargs)
-        self.root_path = kwargs.get('rootPath')
+        if 'rootUri' in kwargs:
+            self.root_uri = kwargs.get('rootUri')
+        elif 'rootPath' in kwargs:
+            root_path = kwargs.get('rootPath')
+            self.root_uri = urljoin(u'file:', pathname2url(root_path))
+        else:
+            self.root_uri = ''
```

Comment:
I have to admit ignorance on what kwargs or pathname2url returns here. Not specifying unicode literal works in python3, but in python2 I see the following:
```
self.root_uri = urljoin('file://', pathname2url(root_path))
  File "/Users/tomv/Projects/palantir/python-language-server/.tox/py27/lib/python2.7/site-packages/future/backports/urllib/parse.py", line 418, in urljoin
    base, url, _coerce_result = _coerce_args(base, url)
  File "/Users/tomv/Projects/palantir/python-language-server/.tox/py27/lib/python2.7/site-packages/future/backports/urllib/parse.py", line 115, in _coerce_args
    raise TypeError("Cannot mix str and non-str arguments")
```